core/src/main/java/org/apache/calcite/rel/logical/LogicalFilter.java

```diff
@@ -47,7 +47,7 @@
  * not targeted at any particular engine or calling convention.
  */
 public final class LogicalFilter extends Filter {
-  private final ImmutableSet<CorrelationId> variablesSet;
+  private final Set<CorrelationId> variablesSet;
```

Comment:
is it possible to keep the immutableSet here? generally Immutable should be preferred.

Reply:
Is there any other changes you would like me to make for this?  The set is currently always immutable, just the signature of the member variable is not.